Progress bar initial commit

ehavazli opened the pull request:
Hey guys please test the progress bar and let me know how it looks.
I tested with ariaTSsetup.py and it is working. I can add it to the initialization of the ARIAProduct.py but wanted to check with everybody before going forward. 

- added a new tool *progBar.py*
- GDAL translate output, Stitch settings, Solver, Mapping complete messages  in unwrapStitching.py is supressed.
- Progress bar added to extractProduct.py - export_product function.
 - Progress bar added to tsSetup.py - generate_stack function.

ehavazli committed:
quality review:
-Changes with respect to Codacy review

ehavazli committed:
Codacy review

ehavazli committed:
quality review:
- implement some Codacy reviews

ehavazli committed:
Codacy review suggestions

codacy-badger committed:
Add Codacy badge

ehavazli committed:
Update README.md

sssangha committed:
Adjusted enumeration in for loops, remove need for subprocess.

ehavazli committed:
switched to enumerate

range(len()) switched enumerate

ehavazli committed:
remove sys and pass

remove sys and pass

ehavazli committed:
Codacy review:

- ARIAProduct.py:     removed trailing whitespaces
- productPlot.py:     removed trailing whitespaces
- unwrapStitching.py: removed unused modules (Edge)
- ariaDownload.py:    removed subprocess usage
                      'ASFDataDload.py' file written out and imported within the code to avoid subprocess usage.

ehavazli committed:
removed unused module

unwrapStitching.py: Vertex
ariaDownload.py: subprocess

ehavazli committed:
initial commit of progress bar class

ehavazli committed:
progress bar added to processes

ehavazli committed:
change file mode +x
changed file mod

dbekaert commented:
Can you let us know how it works in verbose mode?

dbekaert commented:
It looks like a lot of your earlier commits re-appear in this PR.

ehavazli commented:
> Can you let us know how it works in verbose mode?

This all the output in both verbose and non-verbose mode
```
(aria) [MT-206092:~/timeseries/test] havazli% ariaTSsetup.py -f '../T04/data/*.nc' -d DEM/SRTM_3arcsec.dem -b polygon_test.msk -nt 10
***Time-series Preparation Function:***
All (438) GUNW products meet spatial bbox criteria.
Group GUNW products into spatiotemporally continuous interferograms.
All (219) interferograms are spatially continuous.
Thread count specified for gdal multiprocessing = 10
Download/cropping DEM

Extracting unwrapped phase, coherence, and connected components for each interferogram pair
Generating: unwrappedPhase - [==================================================] 20190723_20190711  211s /     4s
Generating: coherence - [==================================================] 20190723_20190711   29s /     0s

Extracting single incidence angle, look angle and azimuth angle files valid over common interferometric grid
Generating: incidenceAngle - [==================================================] 20150825_20150310 
Generating: lookAngle - [==================================================] 20150825_20150310 
Generating: azimuthAngle - [==================================================] 20150825_20150310 
Directory ./stack already exists.
Number of unwrapped interferograms discovered:  219
[==================================================] 20190605_20190430    1s /     0s
unwrapStack : stack generated
Directory ./stack already exists.
Number of coherence discovered:  219
[==================================================] 20190605_20190430    4s /     0s
cohStack : stack generated
Directory ./stack already exists.
Number of connectedComponents discovered:  219
[==================================================] 20190605_20190430    1s /     0s
connCompStack : stack generated
```

dbekaert commented:
Look pretty nice, we should do some testing.
- What is the pair after the progress bar?
- @sssangha we should check for the verbose mode, I understood that there was more output to be given.

dbekaert commented:
@ehavazli can you clean up the commit history as well.

ehavazli commented:
>     * What is the pair after the progress bar?

@dbekaert the pair name is updated while being processed in the code so it changes with the percentage.

I will clean up the previous commits I have no ide why merged commits appeared here again.

ehavazli committed:
Update progBar.py

ehavazli committed:
Update productPlot.py

ehavazli commented:
I think I managed it but please check again if I am changing something unrelated to progress bar.

dbekaert commented on tools/ARIAtools/extractProduct.py:
```diff
@@ -339,14 +340,23 @@ def export_products(full_product_dict, bbox_file, prods_TOTbbox, layers, dem=Non
 
         # Iterate through all IFGs
-        print("Generating: " + key)
+        # print('Generating: ' + key)
```
if commented out and no longer needed than perhaps provide more as a comment or remove it.

dbekaert commented on tools/ARIAtools/extractProduct.py:
```diff
@@ -339,14 +340,23 @@ def export_products(full_product_dict, bbox_file, prods_TOTbbox, layers, dem=Non
+        ##Progress bar
+        from ARIAtools import progBar
+        prog_bar = progBar.progressBar(maxValue=len(product_dict[0]),prefix='Generating: '+key+' - ')
+        ##############
```
not sure we need the #########

dbekaert replied:
Can be done elsewhere as well.

dbekaert commented on tools/ARIAtools/extractProduct.py:
```diff
@@ -645,7 +653,7 @@ def tropo_correction(full_product_dict, tropo_products, bbox_file, prods_TOTbbox
                 lookfile=gdal.Open(os.path.join(outDir,'lookAngle',product_dict[2][i][0])).ReadAsArray()
             else:
-                lookfile=gdal.Open(os.path.join(outDir,'lookAngle',product_dict[2][0][0])).ReadAsArray()
+                lookfile=gdal.Open(os.path.join(outDir,'lookAngle',product_dict[2][i][0])).ReadAsArray()
```
I think this is undoing an earlier fix. check with @sssangha

sssangha replied:
Yeah, this shouldn't be changed @ehavazli. This was a bug @bbuzz31helped isolate. The idea is that either a user extracts look angles for each pair (as is done currently through ariaExtract) or passes one valid nanmean averaged look angle file (as is done through ariaTSsetup). So that is to say, for ariaTSsetup, there would only be one look angle file passed in the dictionary, and this conditional is meant to capture that.

ehavazli replied:
this doesn't look like something I did with the progress bar but probably missed while fixing all the other commits. I will edit this.

dbekaert commented on tools/ARIAtools/progBar.py:
```diff
@@ -0,0 +1,104 @@
+import numpy as np
+import time
+import sys
+###########################Simple progress bar######################
```
not needed

dbekaert commented on tools/ARIAtools/progBar.py:
```diff
@@ -0,0 +1,104 @@
+import numpy as np
```
Add the same header like we do for all our programs.

dbekaert commented on tools/ARIAtools/progBar.py:
```diff
@@ -0,0 +1,104 @@
+        of future statements."""
+        if self.print_msg:
+            print(' ')
+################################End of progress bar class####################################
```
can be removed, we do not do this in our codes

dbekaert commented on tools/ARIAtools/unwrapStitching.py:
```diff
@@ -517,7 +516,7 @@ def __calculateCyclesOverlap__(self):
         self.fileMappingDict = fileMappingDict
 
-        print('MAPPING complete:')
+        # print('MAPPING complete:')
```
i think there should still be a verbose mode.@sssangha please add your thoughts.

sssangha replied:
Yup, I agree these unwrapStitching progress prints should remain in verbose. Will have to think about similar print statements in the other functions that should also be restricted to verbose mode.

ehavazli replied:
these messages may be included in the verbose mode so I am not going to remove them from here just yet

dbekaert commented on tools/ARIAtools/unwrapStitching.py:
```diff
@@ -1441,8 +1440,8 @@ def product_stitch_overlap(unw_files, conn_files, prod_bbox_files, bbox_file, pr
     # report method to user
-    print('STITCH Settings: Product overlap approach')
-    print('Solver: Minimize overlap')
+    # print('STITCH Settings: Product overlap approach')
```
See my comment on the verbose mode. @sssangha add comments

sssangha replied:
Yup, I agree!

ehavazli replied:
these messages may be included in the verbose mode so I am not going to remove them from here just yet

dbekaert replied:
perfect, thanks!

dbekaert commented on tools/ARIAtools/progBar.py:
```diff
@@ -0,0 +1,104 @@
+        self.prefix = prefix
+
+        self.print_msg = print_msg
+        ## calculate total width based on console width
```
would clean up unused code or add in comments where needed.

ehavazli committed:
Update extractProduct.py

ehavazli committed:
progress bar updates

- made changes with respect to reviews.
- cleaned some of the commented out parts

ehavazli committed:
Merge remote-tracking branch 'upstream/dev' into dev

dbekaert reviewed (approved):
Thanks for this commit Emre.
I think this is a great new feature!

Lets merge and let people test it more.
We can work on logging and the verbose in a next PR.
